proposals/LP0004-project-governance.md

```diff
@@ -0,0 +1,393 @@
+
+We expect _public communication_. As an increasingly large and dispersed organization we rely more and more on public and asynchronous communication.
+
+We value _community over code_. We believe that a healthy and strong community can overcome any obstacle, code or otherwise. For this reason we will never sacrifice the health of our community for a technical contribution.
+
+We embrace a _broad definition of contribution_. Throughout this document we talk about contributions and community involvement. The definition of _contribution_ is not limited to just code commits. We value and recognize the many shapes that _contributions_ take.
+
+### Consensus-Seeking Decision Making
+
+The spirit of [LP0001 LLVM Decision Making](https://github.com/llvm/llvm-www/blob/main/proposals/LP0001-LLVMDecisionMaking.md) is to provide a framework to help the community arrive at a decision around contentious topics. As described the process does not follow a fully democratic process and it acknowledges the possibility that consensus cannot be reached. These are strengths of that proposal process and nothing in this proposal is intended to weaken that.
+
+Formal consensus processes (i.e. voting, surveys, etc) are _extremely_ difficult to execute on decentralized projects like LLVM, and not necessary for healthy community growth. Formal consensus is not a goal.
+
+Consensus-seeking decision making instead focuses on efforts to build consensus and address concerns. It does not require that all concerns are addressed, nor that a full consensus is reached.
+
+This proposal gives decision making authority to project governance structures defined below.
+
+
+###  Role of Code Owners
+
+This proposal does not change the role of code owners as defined in the [LLVM Developer Policy](https://llvm.org/docs/DeveloperPolicy.html#code-owners). As stated in the policy:
+
+    The sole responsibility of a code owner is to ensure that a commit to their area of the code is appropriately reviewed, either by themself or by someone else.
+
+
+### Code of Conduct
+
+All participants in the LLVM Community must follow the LLVM Community Code of Conduct. That applies to the representatives of governance bodies just as it does to any other individual in the community. The Code of Conduct Committee can suspend or revoke the privilege of any individual to participate in the LLVM Governance process for violations of the Code of Conduct.
+
+Misconduct in elections or governance roles such as fraudulent voting are violations of the Code of Conduct.
+
+
+### Voting Contributors
+
+The first structure is to define the _voting contributor_ base. The _voting contributor_ base seeks to represent the active project contributors. In the context of project governance, the only responsibility of a _voting contributor_ is to vote to elect members of the _area teams_.
```

Comment:
I don't think this is a solvable problem, nor do I think we should try to. Each person coming to the community whether they are employed by a company, a hobbyist, or from academia are going to be biased by their own goals with the project. Singling out that corporate contributors may have company-endorsed agendas isn't really helpful because a hobbyist would have their own agenda, and an academic theirs. What makes a company-endorsed agenda inherently worse?

Given that a large percentage of our community is paid contributors and they account for a huge swath of the end users of LLVM it seems that limiting the voice of corporate contributors would be detrimental to the project and community.

In a separate conversation someone commented about the possibility of a single corporate entity manipulating voting by staffing up a large number of eligible voters, and IMO we should welcome that. If a corporate contributor wants to staff 500 engineers to work on LLVM so they get a bigger voice in the community I don't see how that harms LLVM in the long run.

This all biased by my own experience of being a corporate contributor to LLVM. I've definitely had times when I felt that my employer's will for the project wasn't in the best interests of the project or community, but that has never stopped me from putting the community first.

Reply:
Please consider the use of corporate/company in terms of a big organizations of people having a common goals, funding and many contributors pursuing it.

> If a corporate contributor wants to staff 500 engineers to work on LLVM so they get a bigger voice in the community I don't see how that harms LLVM in the long run.

I respectfully disagree with this and LLVM has had rough rides in such cases. The community should probably consider that in some way. These 500 engineer could easily vote out diverse use-cases that enable usage in other contexts.

 > This all biased by my own experience of being a corporate contributor to LLVM. I've definitely had times when I felt that my employer's will for the project wasn't in the best interests of the project or community, but that has never stopped me from putting the community first.

I guess that was the motivation for my comment as some people might prioritize this differently.